rust: reproducibility issue fix with v1.75

With 1.75 rust release, the '.rustc' section of shared object libs are embedded with absolute path names which is casuing reproducibiluty issues.
This change will fix the path name format back to '/rust/$hash' as in earlier versions.

Below are the links for detailed bug description & discusssion with upstream rust.
https://github.com/rust-lang/rust/issues/120825#issuecomment-1964307219
https://github.com/rust-lang/rust/issues/120825#issuecomment-1964652656

Upstream-Status: Inappropriate [patches need rework]
Signed-off-by: Sundeep KOKKONDA <sundeep.kokkonda@windriver.com>
---
--- a/compiler/rustc_session/src/session.rs	2023-12-21 08:55:28.000000000 -0800
+++ b/compiler/rustc_session/src/session.rs	2024-02-26 07:29:15.527577022 -0800
@@ -1269,7 +1269,7 @@
                 | CrateType::Rlib
                 | CrateType::Staticlib
                 | CrateType::Cdylib => continue,
-                CrateType::ProcMacro => return false,
+                CrateType::ProcMacro => return true,
             }
         }
 
